stdlib/public/TensorFlow/Gradients.swift

```diff
@@ -540,17 +540,28 @@ extension Tensor where Scalar : TensorFlowFloatingPoint {
   ) -> (Tensor, (Tensor) -> Tensor) {
     let value = reshaped(toShape: newShape)
     return (value, { v in
-      return v.reshaped(toShape: self.shapeTensor)
+      return v.scalarCount == 1 ?
```

Comment:
Please revert this change.

Shape mismatches were caused by calling `gradient` on functions that don't return scalar tensors. Instead of adding a special derivative definition that performs broadcasting, the correct change is to call `pullback` with a direction vector of the correct shape (same shape as function result).

Let me know if this makes sense!

Reply:
Ah, okay. I was misunderstanding how gradient could be used.